toolchest_client/tools/tool.py

```diff
@@ -310,7 +319,15 @@ def _kill_query_threads(self):
             if thread.is_alive() and thread_status not in {ThreadStatus.COMPLETE, ThreadStatus.FAILED}:
                 self.query_thread_statuses[thread_name] = ThreadStatus.INTERRUPTING
 
-        self._wait_for_threads_to_finish(check_health=False, wait_for_upload=False)
+        # Wait for remaining threads to finish terminating
+        increment_seconds = 5
+        for thread in self.query_threads:
+            while thread.is_alive():
+                self._pretty_print_pipeline_segment_status()
+                self.elapsed_seconds += increment_seconds
```

Comment:
Why was `elapsed_seconds` moved to a class attribute instead of keeping it local to this function?

Reply:
We also use `elapsed_seconds` in `_kill_query_threads`. (In fact, we were resetting `elapsed_seconds` to 0 before.)

Reply:
Un-resolving this thread – `_kill_query_threads` is called from this function, could it be passed to `_kill_query_threads` as a function parameter instead of a class attribute?

Reply:
Ah I see, it's used several other places. Checks out